UnityProjectDP/Assets/Scripts/Visualization/UI/PopUps/AbstractPopUp.cs

```diff
@@ -1,26 +1,40 @@
-﻿using TMPro;
+﻿using AnimArch.Extensions;
+using AnimArch.Visualization.Diagrams;
 using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.UI;
 
 namespace AnimArch.Visualization.UI
 {
     public abstract class AbstractPopUp : MonoBehaviour
     {
         public GameObject panel;
-        public TMP_InputField inp;
-        protected TMP_Text className;
 
-        public virtual void ActivateCreation(TMP_Text classTxt)
+        protected void ChangeStateOfButtons(bool state)
         {
+            DiagramPool.Instance.ClassDiagram.graph.GetComponentsInChildren<GraphicRaycaster>()
+                .ForEach(x => x.enabled = state);
+            var parent = transform.parent.parent;
+            parent.Find("RightMenu").GetComponentsInChildren<Button>()
+                .ForEach(x => x.enabled = state);
+            parent.Find("TopMenu").GetComponentsInChildren<Button>()
+                .ForEach(x => x.enabled = state);
+            parent.Find("TopMenu").GetComponentsInChildren<EventTrigger>()
+                .ForEach(x => x.enabled = state);
+        }
```

Comment:
neviem uplne presne, preco sa to tu robi ale input parameter by som nazval skorej `enabled` ako state (ak to sedi) a rozbil by som to na jednotlive metody podla toho ci je to right menu, top menu ...

Reply:
Spraví to buttony neaktívne ak je zobrazený popUp. Prišlo mi to lepšie dať všetko naraz do jednej metódy ako urobiť 3 samostatné metódy